apps/client/src/module/auth/register-form.tsx

```diff
@@ -67,17 +70,42 @@ export function RegisterForm() {
   async function action(payload: RegisterPayload) {
     const res = await register(payload);
 
-    if (res?.error) {
-      setError("root", { message: res.error });
+    if (res?.success) {
+      setIsSuccess(true);
+    } else if (res?.error) {
+      setError("root", { message: getErrorMessage(res.error) });
     }
   }
 
+  function getErrorMessage(error: string): string {
+    switch (error) {
+        case "account already exists":
+            return t("RegisterPage.error.alreadyExists");
+        case "registration failed":
+            return t("RegisterPage.error.registrationFailed");
+        case "Email verification failed":
+            return t("RegisterPage.error.emailSendingFailed");
+        default:
+            return t("RegisterPage.error.unknownError");
+    }
+}
+
   const onSubmit = handleSubmit((data) => {
     console.log(data);
     const { confirmPassword, realName, mobile, ...filteredData } = data;
     action(filteredData);
   });
 
+  if (isSuccess) {
+    return (
+      <Successed
+        title={t("RegisterPage.succeed.congratulation")}
```

Comment:
if register failed, we want to keep the behaviour as we did, showing failed msg at form for user to easily refill it not redirect it to a new page